tidy3d/plugins/near2far/near2far.py

```diff
@@ -518,105 +559,185 @@ def fields_spherical(self, r, theta, phi):
         )
 
         # assemble E felds
-        E_theta = -scalar_proj_r * (L_phi + eta * N_theta)
-        E_phi = scalar_proj_r * (L_theta - eta * N_phi)
-        E_r = np.zeros_like(E_phi)
-        E = np.stack((E_r, E_theta, E_phi))
+        Et_array = -scalar_proj_r * (L_phi + eta * N_theta)
+        Ep_array = scalar_proj_r * (L_theta - eta * N_phi)
+        Er_array = np.zeros_like(Ep_array)
 
         # assemble H fields
-        H_theta = -E_phi / eta
-        H_phi = E_theta / eta
-        H_r = np.zeros_like(H_phi)
-        H = np.stack((H_r, H_theta, H_phi))
+        Ht_array = -Ep_array / eta
+        Hp_array = Et_array / eta
+        Hr_array = np.zeros_like(Hp_array)
+
+        dims = ("r", "theta", "phi")
+        coords = {"r": [r], "theta": theta, "phi": phi}
+
+        Er = xr.DataArray(data=Er_array[None, ...], coords=coords, dims=dims)
```

Comment:
Does the `None` add an index for `r`?  Just trying to understand

Reply:
Yeah exactly. The fields are initially computed as two-dimensional arrays (in theta and phi), for a fixed r, but I figured it makes sense also to have a record of that r value in the dataset.